design/src/server/anatomy.md

```diff
@@ -318,40 +319,13 @@ stateDiagram-v2
 
 Note that the `S` and `L` are specified by logic written, in Rust, by the customer, whereas `Upgrade`/`UpgradeLayer` is specified entirely by Smithy model via the protocol, [HTTP bindings](https://awslabs.github.io/smithy/2.0/spec/http-bindings.html), etc.
 
-The procedure of taking a struct and transforming it into a HTTP service is formalized by the [`Upgradable`](https://github.com/awslabs/smithy-rs/blob/4c5cbc39384f0d949d7693eb87b5853fe72629cd/rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs#L222-L229) trait:
+The procedure of taking a struct and transforming it into a HTTP service is formalized by the [`Upgradable`](https://github.com/awslabs/smithy-rs/blob/9a6de1f533f8743dbbc3fa6ad974d104c8b841f4/rust-runtime/aws-smithy-http-server/src/operation/upgrade.rs#L220-L225) trait:
 
 ```rust
-impl<P, Op, S, L> Upgradable<P, Op> for Operation<S, L>
-where
-    // `Op` is used to specify the operation shape
-    Op: OperationShape,
-
-    // Smithy input must convert from a HTTP request
-    Op::Input: FromRequest<P>,
-    // Smithy output must convert into a HTTP response
-    Op::Output: IntoResponse<P>,
-    // Smithy error must convert into a HTTP response
-    Op::Error: IntoResponse<P>,
-
-    // The signature of the inner service is correct
-    S: Service<Op::Input, Response = Op::Output, Error = Op::Error> + Clone,
-
-    // The modified Layer applies correctly to `Upgrade<P, Op, S>`
-    L: Layer<Upgrade<P, Op, S>>,
-
-    // The signature of the output is correct
-    L::Service: Service<http::Request<B>, Response = http::Response<BoxBody>>,
-{
-    type Service = L::Service;
-
-    /// Takes the [`Operation<S, L>`](Operation), then applies [`UpgradeLayer`] to `S`, then finally applies the `L`.
-    ///
-    /// The composition is made explicit in the method constraints and return type.
-    fn upgrade(self) -> Self::Service {
-        let Operation { inner, layer } = self;
-        let layer = Stack::new(UpgradeLayer::new(), layer);
-        layer.layer(inner)
-    }
+/// An interface to convert a representation of a Smithy operation into a [`Route`].
+pub trait Upgradable<Protocol, Operation, ...> {
```

Comment:
NIT: We don't use "..." elsewhere, we just pretend like they don't exist at all.

Reply:
Let's keep it consistent - amended in https://github.com/awslabs/smithy-rs/pull/1954/commits/b3de7a412bbb53dae4a52f9673acabec8867547b